### ISSUE TITLE
[EPIC] Pinning user experience

### ISSUE THREAD

**Issue by meiqimichelle** [ipfs/user-research#16]
The purpose of this issue is to track progress on the 2019 Q4 endeavor around the user experience of pinning on IPFS and via the IPFS protocol.

//cc @momack2

**Comment by meiqimichelle** [comment]
**Today**
- [P0] Booted up Github repo (rewrote home README to reflect current status of this work and old work; transferred important points from statement of work -- slightly rewritten for an external audience; created new work tracking structure to make it easier for folks to follow along -- a pinning ux directory, label to filter by, and this EPIC issue; created issues for the next stage of work on this endeavor)
- [P0] Drafted list of entry points points for @momack2 review: https://gist.github.com/meiqimichelle/879a9b7ba89f441880dac656a43d2c61 . This is based on a conversation and braindump from @hsanjuan as a starting point.
- [P0] Connected with @warpfork to deep-dive on IPLD selectors over multiple hours. Notes here: https://gist.github.com/meiqimichelle/7f2c300eb83f3f29a1ce2c6ad4c2df46
- [P1] Started documenting initial user experiences of a few of the listed entry points (IPFS Companion and IPFS Desktop utility (not full WebUI/Desktop yet). See https://gist.github.com/meiqimichelle/1f766a67b338852bde895b9053f14fc6 and https://gist.github.com/meiqimichelle/421e5987b9c2e0925742936a7b61e2b6
- [P1] Announced new project focus in Slack so folks know not to look to me for older assignments, and know where to go instead

**Tomorrow**
- [P0] Continue fleshing out and getting feedback on the entrypoint list
- [P0] Continue documenting user experience of "pins" via each entry point
- [P0] Clean up and share notes from conversation with @warpfork
- [P1] Continue finding existing research to add to #14 
- [P1] Get feedback from stakeholders as available and possible (async or sync)

**Notes**
- This was a long work day. The "today" list will not always be this long. Friday I have a few commitments during the day, so I want to make the most of today (Weds) and tomorrow (Thurs).

**Comment by meiqimichelle** [comment]
**Today**
- [P0] 1:1 with @momack2 for feedback on progress so far (notes in an internal 1:1 document)
- [P0] Based on feedback ^^, moved more towards initial user journey definition where possible (rather than deep dives in any one particular complex corner of our technology). See https://gist.github.com/meiqimichelle/73ab20e869234a81ed3925875d93e223 for an initial exploration into multiple app + WebUI pinning.
- [P0] Updated entrypoint list with info gathered async from @lidel and @hacdias. Will do deeper dive with @lidel tomorrow.
- [P0] Reached out to more stakeholders, this time focusing on a few external-to-PL folks.
- [P1] Summarized and re-wrote notes from my conversation with @warpfork, which I've shared with him for further clarification. They're not pubic yet. Given feedback above, this task fell a little further down on the priority list. The notes might instead just make their way in bits and pieces into other docs.

**Tomorrow**
- [P0] High-bandwidth sync with @lidel 
- [P0] Continue documenting concrete pinning user journeys, a la https://gist.github.com/meiqimichelle/73ab20e869234a81ed3925875d93e223 (may not be visual, but the point is...more in this direction, less in loose deep dives into technical topics). In particular, need to explore the CLI/API side of the pinning experience more. I also want to document a large data journey.
- [P1] Do something with the raw notes at https://gist.github.com/meiqimichelle/1b0cec81da0befc85e11af916b4a760d
- [P1] Continue adding existing research to add to #14 . In particular: need to look and see if old DDS and Cube work, and Cluster research docs, have some insight.

**Notes**
- This bullet is intentionally left blank.

**Comment by meiqimichelle** [comment]
**Today**
- [P0] Sync with @lidel on pinning UX + Companion + other GUI properties
- [P1] In the process of organizing a sync with non-PL pinning API user

**Tomorrow/Monday**
- [P0] Revisit Companion-ish user journey with new info from @lidel (ie, iterate on https://gist.github.com/meiqimichelle/73ab20e869234a81ed3925875d93e223)
- [P0] Explore the CLI/API side of the pinning experience more. 
- [P0] Document a large volumes data journey. See if old DDS and Cube work, and Cluster research docs, have some insight.
- [P2] Consider closing #14 . This work is taking place in other issues.
- [P2] Do something with the raw notes at https://gist.github.com/meiqimichelle/1b0cec81da0befc85e11af916b4a760d

**Notes**
- As noted in https://github.com/ipfs/user-research/issues/16#issuecomment-537788844, my Friday had several non-work commitments that could not be moved. This day was shorter than Wed and Thurs.

**Comment by meiqimichelle** [comment]
**Today**
- [P0] Started draft PRD with user motivations+workflows (for now, in PL-only doc just to make commenting easier -- will try to move soon! https://docs.google.com/document/d/1UVC5BVa7rHYNs359RpN1smPN4FUdZTqvIoDMm0pJ--0/edit#heading=h.ddztyzip7s58 ). So far, have started flows for a "John Q. Curious Public"; an "Alexandria Librarian"; and a "Hacker MacDev".
- [P0] Revisited Companion-ish user journey; not finished yet, and also seeing if I can get some feedback from @lidel (he may not have time)
- [P2] In the process of organizing a sync with non-PL pinning API users

**Tomorrow**
- [P0] Sync with @momack2 
- [P0] Explore the CLI/API side of the pinning experience more. 
- [P2] Consider closing #14 . This work is taking place in other issues.
- [P2] Do something with the raw notes at https://gist.github.com/meiqimichelle/1b0cec81da0befc85e11af916b4a760d

**Notes**
- I wasn't feeling great this morning and so got less done than I'd hoped. Hopefully tomorrow I will be back to full energy.
- Reminder: I will be OOO starting in the evening Tuesday, and will be back at the keyboard Thursday morning.

**Comment by meiqimichelle** [comment]
**Today**
- [P0] Continued draft PRD with user motivations+workflows, in particular responding to in-doc comments from @momack2. v1 of top 3 user workflows ready for review.
- [P0] Sync with @momack2 
- [P1] Publish gist with notes re the relationship between pinning and MFS: https://gist.github.com/meiqimichelle/1e4601b4418bf4f46007f4777aff395d
- [P2] In the process of organizing a sync with non-PL pinning API users

**Tomorrow/Thursday**
- [P0] Explore the CLI/API side of the pinning experience more. 
- [P1] Revisit Companion-ish user journey
- [P2] Consider closing #14 . This work is taking place in other issues.
- [P2] Do something with the raw notes at https://gist.github.com/meiqimichelle/1b0cec81da0befc85e11af916b4a760d

**Notes**
- Reminder: I will be AFK starting in the evening tonight, and will be back at the keyboard Thursday morning.

**Comment by meiqimichelle** [comment]
**Today**
- [P0] Continued draft PRD, in particular responding to in-doc comments from @momack2.
- [P0] Sync with @momack2 
- [P1] Cleaned up notes from sync with non-PL pinning API user

**Tomorrow**
- [P0] Share tl;dr version of three user workflows in this update, to start moving information from the PRD out of docs and into GH
- [P0] Start drafting success metrics for this work, to include plans for any logging, testing, or other feedback mechanisms that should be built
- [P1] Folks have left comments on various artifacts over the last week while I was AFK. Respond to those and/or pull them into synthesis
- [P1] Explore the CLI/API side of the pinning experience more
- [P2] Do something with the raw notes at https://gist.github.com/meiqimichelle/1b0cec81da0befc85e11af916b4a760d

**Notes**
- I had some surprise personal commitments come up that had me AFK for one week. I am back now. Thank you for your understanding.

**Comment by meiqimichelle** [comment]
**Friday/Today**
- [P0] Finished first draft of two of three user workflows. Sharing tl;dr in this update, to start moving information from the PRD out of docs and into GH (⬇️⬇️⬇️see links at bottom of this comment ⬇️⬇️⬇️)
- [P0] Started drafting success metrics for this work, to include plans for any logging, testing, or other feedback mechanisms that should be built
- [P1] Got feedback from non-PL pinning API user on notes; corrections were applied; these notes will stay private as thematic references for now.

**Tomorrow**
- [P0] Finish first draft of final of three user workflows. Share tl;dr link in this update.
- [P0] Sync with @momack2 
- [P1] Folks have left comments on various artifacts over the last week while I was AFK. Respond to those and/or pull them into synthesis

**Notes**
- n/a

**tl;dr versions of two of three user workflows available...**
_"John Q. Curious Public wants to stay in control of his data, and share pictures with his family": https://github.com/ipfs/user-research/issues/17#issuecomment-544327627
_"Data Architect Maria needs to replicate huge datasets across many data centers to make it highly available for scientific study": https://github.com/ipfs/user-research/issues/17#issuecomment-544331471

**Comment by meiqimichelle** [comment]
**Friday/Today**
- [P0] Finished first draft of final of three user workflows (re: data persistence <> pinning services/community co-hosting). Sharing tl;dr in this update (⬇️⬇️⬇️see link at bottom of this comment ⬇️⬇️⬇️)
- [P0] Synced with @momack2 
- [P0] Started responding to comments left various artifacts over the last week while I was AFK.

**Tomorrow**
- [P0] Continue responding to comments on various artifacts. Pull them into synthesis as appropriate.
- [P1] Start drafting current state <> future state gap analysis + overall definition of endeavor success
- [P1] Start drafting chunked feature/milestone plan, hopefully against hypothesis/definition of success

**Notes**
- There's another sync scheduled with a non-PL pinning API user -- Wednesday. Yay!

**tl;dr version of final of three user workflows available...**
_"John Q. Curious Public wants his photos to persist on the distributed web": https://github.com/ipfs/user-research/issues/17#issuecomment-544785574

**Comment by meiqimichelle** [comment]
**Today**
- [P0] Started drafting current state <> future state gap analysis + overall definition of endeavor success
- [P0] Synced with a third-party IPFS API user
- [P1] Continued responding to comments

**Tomorrow**
- [P0] Continue drafting current state <> future state gap analysis + overall definition of endeavor success
- [P0] Sync with @momack2 
- [P1] Start drafting chunked feature/milestone plan, hopefully against hypothesis/definition of success
- [P2] Continue responding to comments on various artifacts.

**Notes**
- Yesterday was basically a wash :(

**Comment by meiqimichelle** [comment]
**Today**
- [P0] Continued drafting current state <> future state gap analysis + overall definition of endeavor success
- [P0] Continued drafting chunked feature/milestone plan
- [P0] Synced with @momack2

**Tomorrow**
- [P0] Ship v0 for @momack2 review of chunked feature/milestone plan
- [P0] Continue drafting overall definition of endeavor success
- [P0] Refine current state <> future state gap analysis

**Notes**
- Goal: share v0 entire PRD next week for broader comment/feedback.

**Comment by meiqimichelle** [comment]
**Today**
- [P0] Continued drafting current state <> future state gap analysis + overall definition of endeavor success
- [P0] Shipped v0.0 for @momack2 review of chunked feature/milestone plan

**Tomorrow/Monday**
- [P0] Continue drafting chunked feature/milestone plan
- [P0] Continue drafting overall definition of endeavor success
- [P0] Refine current state <> future state gap analysis

**Notes**
- I am changing time zones this weekend -- heading to summer! :D See ya on the flip side. Literally.

**Comment by meiqimichelle** [comment]
**Today/Yesterday**
- [P0] Shipped v0.0 current state of IPFS GUI apps for @momack2 review (other "current state" parts shipped earlier and are in refinement stages; GUI was the only part left that hadn't had much attention) (Also forwarded to @lidel for feedback, if he has time)
- [P0] Shipped v0.0 overall definition of endeavor success for @momack2 review
- [P0] Continued working on chunked feature/milestone plan based on @momack2 comments

**Tomorrow**
- [P0] Finish v1 chunked feature/milestone plan
- [P0] Respond to @momack2 comments

**Notes**
- n/a

**Comment by meiqimichelle** [comment]
**Today**
- I'm going to be moving on to other work. Today, I am double-checking that all existing work that should be in this repo is in this repo, and will leave a final status comment when that's done.

**Comment by meiqimichelle** [comment]
**Closeout**

- Summary of background / current status / painpoints of pinning in the IPFS ecosystem: https://gist.github.com/meiqimichelle/25d2221c08807841be7764b22f4a5bb5
- For anyone at IPFS picking up this work in the future, this may be a useful place to start: WIP ["IPFS Pinning UX PRD"](https://docs.google.com/document/d/1UVC5BVa7rHYNs359RpN1smPN4FUdZTqvIoDMm0pJ--0/edit?usp=sharing) draft (Protocol Labs internal access only). Here's a [pdf of the unfinished PRD* as of 30 Oct 2019](https://github.com/ipfs/user-research/files/3824308/IPFS-Pinning-UX-PRD_Draft_30Oct2019.pdf), in case that's helpful for folks outside IPFS researching this area.
- To read about or contribute to this work, you can search for [issues tagged `[Area] Pinning UX`](https://github.com/ipfs/user-research/issues?utf8=%E2%9C%93&q=label%3A%22%5BArea%5D+Pinning+UX%22+) or spelunk in this repo's [`pinning-ux` directory](/pinning-ux).  

\* product requirements document
